lib/hawkular/base_client.rb

```diff
@@ -174,14 +174,16 @@ def tenant_header
     end
 
     def handle_fault(f)
+      http_code = (f.respond_to?(:http_code) ? f.http_code : 0)
```

Comment:
when would this not have http code?
also, if http_code would be set to 0, the exception on line 186 will be confusing.